.eslintrc.js

```diff
@@ -66,4 +66,13 @@ module.exports = {
     'no-plusplus': 'off',
     strict: 'off',
   },
+  overrides: [
+    {
+      // Disable default-props for TypeScript files; it's fine to have undefined prop values if declared that way
+      files: ['**/*.tsx'],
```

Comment:
Definitely in favor of disabling/modifying this rule, especially as we adopt more TypeScript stuff. Though, I do wonder if it's worth disabling the rule for both `.jsx` and `.tsx`, given `defaultProps` for functions are technically deprecated and are expected to be removed in React 19. Non-TypeScript files could also probably incrementally start migrating to ES6 default function parameters today as well.

If we were to disable the rule across all `.jsx` and `.tsx` files now, would this config be moved under the `rules` property instead of `overrides`?

---

[suggestion] Also, I wonder if it'd be worth considering whether to _modify_ the rule versus _disabling_ it altogether. For example, I imagine we might still want linting on missing `defaultProps` for any class components that might still exist (where ES6 default function params aren't applicable?). Similarly, there is a rule option (i.e., `functions`) for changing the expected strategy for defaulting props; by default, it is `defaultProps` but could be changed to `defaultArguments` instead (see [docs](https://github.com/jsx-eslint/eslint-plugin-react/blob/51d342ba350ae7d7dabce1caa648e71926ef283f/docs/rules/require-default-props.md#functions)). Changing the `functions` option to `defaultArguments` seems like it might be preferable versus disabling the rule altogether? 🤔

Reply:
I'd love to just disable the rule altogether.

As for changing to `defaultArguments`, that's fine with me but I think it would cause way too many warnings/errors with existing code which is written using `defaultProps` (unless that's what you want, to force a big migration).

I'd still prefer to disable it altogether in TypeScript files, because I often write code like this (real example), which I think is perfectly fine:

```ts
const SearchUI: React.FC<{ courseId?: string, closeSearchModal?: () => void }> = (props) => {
  ...
```
and yet with `defaultArguments` on, eslint wants me to write it much more verbosely as:

```ts
const SearchUI: React.FC<{ courseId?: string, closeSearchModal?: () => void }> = ({
  courseId = undefined,
  closeSearchModal = undefined,
}) => {
  ...
```

Reply:
@adamstankiewicz Would you like me to update this to just disable the rule altogether? (for JSX and TSX)

Reply:
@adamstankiewicz I updated this to not required `defaultProps` for function components in general, and added a comment explaining why it's deprecated.

I'd love to change the rule to also require `defaultArguments` but I'm not sure how to roll that out without requiring major changes of existing code, so I think this is the best incremental improvement.

Could you please take a look and approve if it's good with you?

Reply:
@bradenmacdonald I will take a look later this evening!